pkg/extensions/brokers/natsjetstream/natsjetstream.go

```diff
@@ -187,13 +199,23 @@ func (c *Controller) HandleMessage(msg jetstream.Msg, sub extensions.BrokerChann
 	}
 
 	// Create and transmit message to user
-	sub.TransmitReceivedMessage(extensions.BrokerMessage{
-		Headers: headers,
-		Payload: msg.Data(),
-	})
-
-	// Acknowledge message
-	_ = msg.Ack()
+	sub.TransmitReceivedMessage(extensions.NewAcknowledgeableBrokerMessage(
+		extensions.BrokerMessage{
+			Headers: headers,
+			Payload: msg.Data(),
+		},
+		AcknowledgementHandler{
+			doAck: func() {
+				if err := msg.Ack(); err != nil {
+					c.logger.Error(ctx, fmt.Sprintf("error on ack message: %q", err.Error()))
+				}
+			},
+			doNak: func() {
+				if err := msg.NakWithDelay(c.nakDelay); err != nil {
```

Comment:
Out of curiosity : is there any reason to use the `WithDelay`function ? :)

Reply:
In your draft you had described a nak with delay, so I included that. However, the nakDelay can also be set to 0 by the user.